### ISSUE TITLE
key maps not working?

### ISSUE THREAD

**Issue by nilsbecker** [nojb/llpp#5]
i tried this file: 

https://gist.github.com/Earnestly/7608794

as `~/.config/llpp.conf` to get vi-like key maps. (i removed the ui font section). llpp starts but then complains 

```
2016-07-12 11:25:29.253 llpp[30771:12119456] After caml_main
error processing attribute ("in"="menu") at 85
Failure("int_of_string")
```
and similar errors on each binding that i tried. the same situation for the keys file distributed with llpp.

after closing llpp, the config file gets completely overwritten with an autogenerated one.

**Comment by moosotc** [comment]
Nicolas Ojeda Bar <notifications@github.com> writes:

> /cc @moosotc
>

I'm not sure what you mean by that.

-- 
mailto:moosotc@gmail.com

**Comment by nilsbecker** [comment]
(i think it means 'carbon copy' indicating that he is bringing the issue to your attention) 

i opened the issue here since i was not sure if this behavior is only seen in the mac version.

**Comment by nojb** [comment]
I don't think this is an issue with the cocoa port...

**Comment by moosotc** [comment]
nilsbecker <notifications@github.com> writes:

> (i think it means 'carbon copy' indicating that he is bringing the
> issue to your attention)

I expected it to have some context, never bothered to go to github to
look at the original filing.

>
> i opened the issue here since i was not sure if this behavior is only
> seen in the mac version.
>

Yes it's unique to cocoa version, by that I mean that: cocoa's wsi.ml
doesn't "decode" the key names to key values, and given Unrecognized
fallback you can only remap rather limited subset of keys.

-- 
mailto:moosotc@gmail.com

**Comment by nojb** [comment]
Ah ok, I missed that.  I will look into it.

**Comment by nojb** [comment]
I just pushed a tentative fix for this in f10b28cdef749be547f2755c9a2ac54f86334484; could you give it a go whenever you can and report back ? I don't actually have a mac nearby so can't test it at the moment.
Thanks!

**Comment by nilsbecker** [comment]
i first tried with a blank config file. the standard keys work including pgup/pgdown which are fn+up/down for me. exception: `delete` to go back a page which i have to access via `fn`+`backspace`.

**Comment by nojb** [comment]
Yes, the `delete <-> fn + backspace` stuff needs a little more work.  I will look into it.
